api/middlewares/auth.go

```diff
@@ -0,0 +1,54 @@
+package middlewares
+
+import (
+  "github.com/gin-gonic/gin"
+  "github.com/charmbracelet/log"
+  "net/http"
+  "strings"
+  "github.com/Salz-Team/salz/api/db"
+  "time"
+  "database/sql"
+)
+
+func AuthMiddleware(authDb db.AuthDBHandler) gin.HandlerFunc {
+    return func(c *gin.Context) {
+        log.Debug("Auth middleware")
+        authTokenHeader := c.GetHeader("Authorization")
+
+        if authTokenHeader == "" {
+            log.Info("No token found")
+            c.Redirect(http.StatusTemporaryRedirect, "/login")
+            c.Abort()
+            return
+        }
+
+        authToken := strings.TrimSpace(strings.Replace(authTokenHeader, "Bearer", "", 1))
+
+        token, err := authDb.GetToken(authToken)
+        if err == sql.ErrNoRows {
+            log.Debug("Token not found (err no rows)")
+            c.AbortWithStatus(http.StatusUnauthorized)
+            return
+        } else if err == db.ErrTokenNotFound {
+            log.Debug("Token not found (err token not found)")
+            c.AbortWithStatus(http.StatusUnauthorized)
+            return
+        } else if err != nil {
+            log.Error("Unable to get token", "error", err)
+            c.AbortWithStatus(http.StatusInternalServerError)
+            return
+        }
+
+        // Is this the right behaviour?
```

Comment:
plz tell me what to do here

Reply:
@japorized

Reply:
We don't have or want a refresh token flow right? In that case, a redirect to `/login` to re-login is a good flow. That's actually a common thing to do even with a refresh token flow, for when even the refresh token has expired.

Reply:
Idk how gin works here but is `c.Abort()` right? Just making sure.

Reply:
it doesn't need to be there. that's my brain on copy-paste autopilot. I'm pretty sure things are handled properly from just the `c.Redirect`